Copy of istio install templates, converted to helm style

costinm opened the pull request:
The files should produce the same output as the original - I only changed the syntax and 
merged some of the config variants that the updateVersion script was generating.

The goal is to gradually converge to a single helm template - however the current helm is still 
too different and hard to compare with the original, and I couldn't get it into the test framework.

I tested it with both istioctl gen-template and helm template - but most people should probably
use 'helm template', for the '--set' and multiple merged values.xml.

linsun commented:
@costinm it is quite hard to read this large change set... is the main work to rename folders/file location?  Or there is more to it?

costinm commented:
Rename + convert the style to helm syntax. You can diff individual files against the original.
Most changes are related to commented-out fields or things that were replaced by sed or 
updateVersions, now all is controlled via values.xml.

Changes I made:
- exclude zipkin-to-stackdriver if the project id is not specified, and replace the commented out
'set project id' with the value.xml setting if present. It wouldn't work otherwise.

- used a single istioTag / istioHub for all - I don't expect we'll have different hubs.

- used a 'security.cacerts' setting to switch between self-signed and existing. Original had 2 files
and you had to apply one or the other. Same for 'security.singleNamespace' - it was a separate config to apply after istio.yaml.

- configMap gets set from values.xml ( instead of sed-ed)

- added the values.xml-based node-port option to ingress - instead of the commented out option.

- sidecarInjector is enabled only if values.xml contain a CA certificate ( would crash/fail otherwise, no point loading it )

Note that values.xml is copied from the other helm directory, but I added the values used by
the release/sed templates at the top. I'm not yet sure how to map them - in particular since istioctl
defines yet another set.

luisyonaldo commented:
Is this going to replace current helm istio chart? The main idea of the chart is make it configurable to any kind of environment or needs, that's why the main concept of the chart is to follow helm best practice. This one is now basically just a plain manifest inside helm template.

Current one with the correct variables should generate same templates. If you tell me the goal then maybe we can go one by one and compare which template is not being generated correctly.

linsun commented:
I think @luisyonaldo is asking the right questions, how this relates to the exiting helm charts, and if we are maintaining 2 charts in separate dirs (the existing one kubernetes/helm and this new one kubernetes/istio-helm).

costinm commented:
This is intended to replace the sed-generated templates that we use - which are used by the automated
tests and is what we ship as recommended configuration. 
It happens to use helm instead of sed - and will be maintained just like we were maintaining and tested just like the previous version.

I don't know how it relates with the other helm - unfortunately the style and options have diverged, and
it become too complex. It is good to have options - and it would be great to converge, consistency is 
good - at least after this PR (and the one converting pilot's templates) we would use a single templating
syntax. 

My primary concern is the divergence of values.xml options - in particular given istioctl defining it's own.
I don't mind if this set of configs has a very narrow set of options (i.e. mostly what we have automated tests for and recommend for users).

sdake commented:
@costinm regarding the approach, I understand this provides immediate relief from the pain of `updateVersion.sh`.  I strongly feel, however, that using this Helm implementation is really no better than using a manifest, and damaging to Helm integration with Istio, and should be a short-term workaround to resolving the updateVersion pain.

I understand your position when looking at the Helm charts and thinking they are too complex.  This is the cost of the ability to manage upgradeability within a Helm chart, along with other oddities like using an injector A for kube 1.7/1.8, or using injector B for 1.9.  To people that understand Helm, the templates make sense.  I know that isn't a lot of solace to the broader Istio community which must maintain the Helm charts, if they are to be the canonical version.

On the divergence of istioctl, from what I can tell, istioctl values.yaml generation is not maintained in accordance with the master Helm chart.  I'm not sure that istioctl model was ever thought through properly and had community contributors willing to maintain it in concert with the Helm charts.  I think the approach of just making a new Helm chart to match istioctl is worse then using the existing Helm chart and modifying istioctl to suit.

That said, here is my proposal (with no specific list of people to do the work :):

* Identify the sea-salt set of Helm configuration options (which are really an API).  This is mostly done with the existing master helm chart.
* Identify a subset of the sea-salt API that the community deems mandatory and trim it as so.
* Modify istioctl to properly process and generate a values.yaml from the master Helm chart.
* Use master helm chart in the gate with istioctl modifications.

@costinm would this solve your concerns?

 cc @sbezverk

luisyonaldo commented:
@costinm is true that current helm chart and normal manifest are too different, in my opinion they were made with different proposes in mind. Current helm chart try too be as much configurable as possible, so it can work with multiple environments, versions and options. At some point it might be even too flexible. I design it following helm best practices, but I don't know istio that much to know when something make sense or not. For example right now, you can have multiple istio versions at the same time, on one or multiple namespaces. In practice, I think this is not needed or even recommended.

Nevertheless, in my opinion, current helm chart can generate same manifest as the ones generated by `updateVersion.sh`. They might have a different syntax and some annotations that are not important outside helm installation, but the logic should be the same as long it receives the right input.

Having said that, is it possible to start a small test that use helm?. Then we can start the migration of all [tests](https://github.com/istio/istio/blob/master/tests/istio.mk#L65-L72) to use helm and support only one template definition. Right now, all changes are done only on manifests and then need to be cherry picked to helm charts. Also, the new helm chart in this PR is just a copy of the current manifests with almost no configuration, which in my opinion make no sense since is the same as current manifests definitions needed to install istio.

sdake commented:
@costinm re @luisyonaldo's question regarding tests, and one of the problems being a lack of tests, I believe fairly immediate relief can be had from just running the results of a parsed master Helm template in the gate.  Are these the only references to the yaml generation you mentioned in your PR: https://github.com/istio/istio/blob/master/tests/istio.mk#L65-L72

I don't see the test case changes in the PR itself.

linsun commented:
I think I'm with @sdake and @luisyonaldo on this one, I would prefer us to leverage the current helm chart to generate the istio yaml (for non helm enabled env) and istioctl gem-deploy.

costinm commented:
There are few problems here:
1. API surface for config - i.e. what Istio officially support (and tests) in terms of install configuration. 
For example changing the service account name or many of the other options in the helm are unlikely
to work and I doubt they have been tested ( since I know cases in the code that relies on particular 
config - in particular for mesh expansion ). This can't be done ad-hoc in helm - we need to support
either istioctl-defined values ( if we go ahead with that), or define them in the api/ package and treat
this as an API. It's fine to have extra knobs - but the core config ( mtls on/off, telemetry, etc) need proper
definition and testing. 

2. General layout of the helm - the current helm has a pretty bad organization ( ingress as main template ?), and too different from the official release. 

3. It is not clear how the helm is maintained - the official templates get change from time to time. My goal with this PR is to remove the old sed based templates, and have the testing infra switch to this one.

4. istioctl - I think it is useful for non-helm users to be able to generate and modify configs, and it's also
useful for those configs to use the gotmpl/helm syntax. If it is not supposed to work or maintained - we should remove the feature, but we can't have it in a non-working state, and I don't think we can require
helm for istio installation.

5. This PR in particular ( which is still WIP - I didn't want to spend too much time rewriting the test infra
and removing the sed-based files given the expected debates :-) - I think it is valuable to use a single
syntax and have consistency ( and upgradeability !) from the official install yaml  and helm. Users should
be able to install istio.yaml from a release, then either use helm or istioctl to modify options and update
the cluster.

It is possible that the current helm could be modified - and obviously the goal is to converge, or at least
make it easy for developers to change both the official install files ( sed or helm based) and the more 
flexible helm. But at the same time - this PR provides a clear improvement compared to the sed, and I don't think blocking the PR because it would create 2 ways to deploy istio is valid - we already have 2,
the official yaml and the super-customizable but untested helm.

costinm commented:
Also: if someone can commit to change the helm, I would be happy to work on other things. 
The main requirements:

- have istioctl generate an istio.yaml which behaves the same (or almost the same) with what we currently ship and is used by the integration tests.

- 'helm template' or istioctl should be used with the same configs and different flags, and the result can be applied on a cluster configured with the official istio.yaml release to modify settings

- better separation between knobs users should configure, and advanced/debug. I'm trying to 
move the core ones ( that are needed for tests or normal uses cases ) to the top level, and to fit in 
the 'features' defined in istioctl. For example the 'tag' flag from istioctl should be reflected in the
generated config (and not require 5 settings, one for each component )

istio-testing commented:
@costinm: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-presubmit.sh | cd56462f12196fd70d54dd99b9944e5ab372dcc1 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/4440/) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

frankbu commented:
I think it's very important to converge on one set of templates. Having multiple templates, regardless of syntax, is a nightmare to maintain. When we approved the PR to add the helm templates, a few months ago, it was on the understanding that the next step would be to use them instead of the sed templates, but I think what @costinm is saying is that they are too complex and therefore can't easily be used to replace the sed ones. That seems to be the core problem that needs to be fixed, rather than introducing another competing set of helm templates.

Would it make sense to take these new helm templates (or something close to them) and use them as the "official" helm chart for now (i.e., replace the other ones which haven't really worked yet, so don't have users depending on them) and then gradually add more functionality from the other ones. It's always easier to start simple and add functionality, then to start with something overly functional and then try to figure out what's actually needed. Just a thought.

istio-testing commented:
@costinm: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-presubmit.sh | 1a12c845e0ac193f1dd5fa075475ab5a012df4e0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/4466/) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-presubmit.sh | 1a12c845e0ac193f1dd5fa075475ab5a012df4e0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/4466/) | `/test istio-presubmit`
prow/istio-unit-tests.sh | 1a12c845e0ac193f1dd5fa075475ab5a012df4e0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-unit-tests/123/) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

sdake commented:
@frankbu the community did start with something very simple.  The order of events (so history is not repeated with the proposal you have made):

* The initial Helm implementation created by @sbezverk were simple and **non-functional**.  CRDs were created after the rest of the system was started.
* This CRD problem led @luisyonaldo to refactor the current Helm implementation to permit creation of the CRDs in the proper order.

It is factually incorrect to state that the current Helm charts do not work.  They currently work and work well with the default configuration options.  E.g. `helm install istio --name istio`.  I believe one thing @costinm has pointed out, and I agree completely with - is the behavior of the Helm charts is undefined if the API configuration options are used (e.g. ``--set`` option to Helm).

The following from my understanding is missing from the existing Helm work:
* A gate.
* A redesign of the Helm values.yaml API to offer core features in one place and non-core features in another.
* A mechanism for istioctl to generate an istio.yaml that renders similarly to the existing yaml code.

If upstream doesn't provide a quality Helm implementation, we run a very high risk of everyone consuming Istio reimplementing their **own** helm chart, creating deployment fragmentation which never leads to a good outcome.

Given what I've seen out of @luisyonaldo during the review process, I'm certain he can get the Helm charts sorted out to community standards while also avoiding said fragmentation.

@costinm - I wasn't proposing to block this review.  I think it is healthy to understand the requirements so folks aren't kept guessing as to what needs to be done next to finish this work in a way that doesn't result in a long-term deployment fragmentation battle.  Also agree Helm can't be the only way to install Istio.

Cheers
-steve

costinm commented:
I agree with both comments - and that is the intent as well. My short term goal - which I think is very 
important - is to move away from the sed templates and syntax with minimum risk.

I don't mind - after the old templates can be removed - to apply the same refactoring, and properly 
separate the services/etc, or even use separate template directories. But the current sed templates are 
what we have been testing and maintaining since 1.0 and are the most stable configuration.

Regarding 'complexity' - the main problem is on values.yaml and the templates used. Replacing the
name of the mixer, pilot,and many other possible customisations in the new templates are guaranteed
to break something.

munnerz commented:
My thoughts on how this should be done - note, I've been through a similar process in one of our projects [cert-manager](https://github.com/jetstack/cert-manager) (which follows a very similar repo design/layout) quite recently. We ended out with:

* One source of truth (a Helm chart): https://github.com/jetstack/cert-manager/tree/master/contrib/charts/cert-manager
  * values.yaml may be partitioned into 'advanced' options (i.e. unsupported changes)

* Two values.yaml files: https://github.com/jetstack/cert-manager/tree/master/hack/deploy
  * `values-istio.yaml` - a values.yaml file configured to generate what we today call `istio.yaml`
  * `values-istio-auth.yaml` - a values.yaml file configure to generate what we today call `istio-auth.yaml`

* A script, `update-deploy-gen.sh`, that runs `helm template` to generate `istio.yaml` and `istio-auth.yaml`: https://github.com/jetstack/cert-manager/blob/master/hack/update-deploy-gen.sh

* A script, `verify-deploy-gen.sh`, that ensures the `istio-auth.yaml` and `istio.yaml` are up to date based on the Helm chart: https://github.com/jetstack/cert-manager/blob/master/hack/verify-deploy-gen.sh

Which results in this directory of generated manifests: https://github.com/jetstack/cert-manager/tree/master/docs/deploy

These are used to deploy cert-manager during our e2e tests.

This provides a number of advantages:

1) Keeps the `istio.yaml` and `istio-auth.yaml` file, which is a fully e2e tested deployment of Istio that is guaranteed to be working

2) Provides users with the raw `values` files needed to deploy Istio in one of these supported and tested configuration, if they want to use Helm. The values file that results in `istio.yaml` is likely to be empty (i.e. just adopts the helm charts default values). The values file that results in `istio-auth.yaml` would likely just set `security.enabled=true`.

3) The chart is tested with a set of known-good configurations.

4) If a developer doesn't update the `istio.yaml` and `istio-auth.yaml` files, CI will fail in a quick-verify stage (by running the verify-deploy-gen.sh script).

5) Saves having to change your test harness - right now your test harness deploys `istio[-auth].yaml` via kubectl. This is still possible, as we check-in the generated deployment manifests. Additional e2e tests that work directory with the chart are possible in future if required, but shouldn't be required initially at least.

Note that here I've only referenced `istio.yaml` and `istio-auth.yaml`, but this could be extended to many others quite easily (e.g. with rbac, without, etc).

I'm happy to put a PR together to demo this, assuming your CI images have Helm installed already 😄

istio-testing commented:
@costinm: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-presubmit.sh | e031ba9ce7b55906c7322b0817bac0010c39aa07 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/4704/) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

mattdelco commented on Makefile:
```diff
@@ -110,6 +110,7 @@ ISTIO_DOCKER_TAR:=${ISTIO_OUT}/docker
 
 GO_VERSION_REQUIRED:=1.9
 
+# TODO: replace this with DOCKER_ISTIO_HUB (20 lines above)
```
DOCKER_ISTIO_HUB => ISTIO_DOCKER_HUB

I added ISTIO_DOCKER_HUB recently when it became known that istioctl was hard-coding a reference to docker.io.  It was late in a release system and I opted for a more specific option for customization.  After some reflection and greater understanding I'm inclined to think that istioctl should not use a separate option name for customizing the hub.  I had in mind to change ISTIO_DOCKER_HUB to HUB (which is mostly just updating this file, release/cloud_builder.sh, and bin/get_workspace_status).  While I like the idea of using a more specific name than just "HUB" or "TAG" (to reduce conflict with other potential uses), I'm afraid there's be a lot of build, test, and dev scripts to update.

istio-testing commented:
@costinm: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-presubmit.sh | ae962edd1f6e27d949cd45c28423df6ff785e2e4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/4800/) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@costinm: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-presubmit.sh | dd42858b60dea6f06121b8447edf85ad3d38280d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/4902/) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-presubmit.sh | dd42858b60dea6f06121b8447edf85ad3d38280d | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/4902/) | `/test istio-presubmit`
prow/istio-unit-tests.sh | bf6c4d27f8e4bada0e33375f8aab6d408cd9aadd | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-unit-tests/836/) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | bf6c4d27f8e4bada0e33375f8aab6d408cd9aadd | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-unit-tests/836/) | `/test istio-unit-tests`
prow/istio-presubmit.sh | bf6c4d27f8e4bada0e33375f8aab6d408cd9aadd | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/5328/) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-presubmit.sh | bf6c4d27f8e4bada0e33375f8aab6d408cd9aadd | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/5328/) | `/test istio-presubmit`
prow/istio-unit-tests.sh | 667f5a6df8d2530998edc48d40e502e27d00c8da | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-unit-tests/842/) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | 667f5a6df8d2530998edc48d40e502e27d00c8da | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-unit-tests/842/) | `/test istio-unit-tests`
prow/istio-presubmit.sh | 667f5a6df8d2530998edc48d40e502e27d00c8da | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/5336/) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-presubmit.sh | 667f5a6df8d2530998edc48d40e502e27d00c8da | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/5336/) | `/test istio-presubmit`
prow/istio-unit-tests.sh | 53f8a41350244788c10135cf9136b606e0fe3fe9 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-unit-tests/846/) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | 53f8a41350244788c10135cf9136b606e0fe3fe9 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-unit-tests/846/) | `/test istio-unit-tests`
prow/istio-presubmit.sh | 53f8a41350244788c10135cf9136b606e0fe3fe9 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/5340/) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | 53f8a41350244788c10135cf9136b606e0fe3fe9 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-unit-tests/846/) | `/test istio-unit-tests`
prow/istio-presubmit.sh | de261456450e241a5a09a46bf4d054c20ebae280 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/5341/) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@costinm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-presubmit.sh | de261456450e241a5a09a46bf4d054c20ebae280 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-presubmit/5341/) | `/test istio-presubmit`
prow/istio-unit-tests.sh | de261456450e241a5a09a46bf4d054c20ebae280 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3193/istio-unit-tests/847/) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

ldemailly commented on bin/init.sh:
```diff
@@ -112,6 +114,15 @@ if [ ! -f ${ISTIO_OUT}/envoy ] ; then
+if [ ! -f /usr/local/bin/helm -a ! -f ${ISTIO_OUT}/helm ] ; then
+    # Install helm. Please keep it in sync with .circleci
+    cd /tmp && \
+        curl -Lo /tmp/helm.tgz https://storage.googleapis.com/kubernetes-helm/helm-${HELM_VER}-linux-amd64.tar.gz && \
```
try vendor ?
let's not add more stuff that only works on linux for no good reason
